sklearnex/preview/ensemble/_forest.py

```diff
@@ -335,27 +516,28 @@ def _onedal_fit_ready(self, patching_status, X, y, sample_weight):
                 input_is_finite = False
             patching_status.and_conditions(
                 [
-                    (
-                        input_is_finite,
-                        "Non-finite input is not suppored.",
-                    ),
+                    (input_is_finite, "Non-finite input is not supported."),
                     (
                         self.monotonic_cst is None,
-                        "Monotonicity constrains are not supported.",
+                        "Monotonicity constraints are not supported.",
                     ),
                 ]
             )
 
         if patching_status.get_status():
-            if sklearn_check_version("1.0"):
-                self._check_feature_names(X, reset=True)
-            X = check_array(
-                X,
-                dtype=[np.float32, np.float64],
-                force_all_finite=not sklearn_check_version("1.4"),
-            )
-            y = np.asarray(y)
-            y = np.atleast_1d(y)
+            if sklearn_check_version("0.24"):
```

Comment:
Looking into the sklearn versions, _validate_data has been used since 0.24 and follows data prep in sklearn needed for patching checks